internal/declarative/v2/reconciler.go

```diff
@@ -139,7 +139,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 				metrics.ManifestRemoveFinalizerWhenSecretGone, metrics.IntendedRequeue)
 		}
 
-		r.Event(obj, "Warning", "ClientInitialization", err.Error())
+		// Suppress the creation of events based on an "unauthorized against the SKR cluster" condition
+		// to not put burst loads onto ETCD when doing credential rotation
+		if util.IsUnauthorized(err) {
+			r.Metrics.RecordRequeueReason(metrics.ManifestSyncUnauthorized, metrics.UnexpectedRequeue)
+		} else {
+			r.Event(obj, "Warning", "ClientInitialization", err.Error())
+		}
+
 		obj.SetStatus(obj.GetStatus().WithState(shared.StateError).WithErr(err))
 		return r.ssaStatus(ctx, obj, metrics.ManifestClientInit, metrics.UnexpectedRequeue)
```

Comment:
I was considering here to directly returning with requeue reason after recording the unauthorized metric. However, I think we need to update the status (even though it is a transient condition that will resolve itself within a couple of mins). If we wouldn't do so and there is a permanent condition leading to unauthorized errors, we would keep the state healthy and wouldn't be easily able to detect that there is an ongoing issue.